api/v1/ytsaurus_types.go

```diff
@@ -789,17 +791,25 @@ const (
 
 type UpdateStatus struct {
 	//+kubebuilder:default:=None
-	State      UpdateState `json:"state,omitempty"`
-	Components []string    `json:"components,omitempty"`
+	State UpdateState `json:"state,omitempty"`
+	// Deprecated: Use updatingComponents instead.
+	Components         []string    `json:"components,omitempty"`
+	UpdatingComponents []Component `json:"updatingComponents,omitempty"`
 	// Flow is an internal field that is needed to persist the chosen flow until the end of an update.
 	// Flow can be on of ""(unspecified), Stateless, Master, TabletNodes, Full and update cluster stage
 	// executes steps corresponding to that update flow.
+	// Deprecated: Use updatingComponents instead.
 	Flow                  UpdateFlow             `json:"flow,omitempty"`
 	Conditions            []metav1.Condition     `json:"conditions,omitempty"`
 	TabletCellBundles     []TabletCellBundleInfo `json:"tabletCellBundles,omitempty"`
 	MasterMonitoringPaths []string               `json:"masterMonitoringPaths,omitempty"`
 }
 
+type Component struct {
+	ComponentName string               `json:"componentName,omitempty"`
+	ComponentType consts.ComponentType `json:"componentType,omitempty"`
+}
```

Comment:
Minor, but I would suggest using Name and Type fields without duplication of the `Component`-part.